### PR TITLE
Get rid of MonadFail from IO

### PR DESCRIPTION
**Pull request by larskuhtz**
Fix incomplete pattern matches in `IO` and get the code base ready for GHC-8.8.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -27,3 +27,22 @@ constraints:
...
       neat-interpolation < 0.4
+
+-- GHC-8.8
+
+package pact
+    ghc-options: -Wwarn
+    haddock-options: --optghc=-Wwarn
+
+allow-newer:
+    *:sbv
```
Comment (emilypi) [inline cabal.project]: this clause may be unnecessary once https://github.com/kadena-io/pact/pull/783 goes in.

File: src/Chainweb/Pact/Backend/ChainwebPactDb.hs [hunk 2]
```diff
@@ -694,11 +698,14 @@ handlePossibleRewind bRestore hsh = do
...
         assign bsBlockHeight bRestore
-        SInt txid <- callDb "getting txid" $ \db ->
+        r <- callDb "getting txid" $ \db ->
           expectSingleRowCol msg =<< qry db
               "SELECT endingtxid FROM BlockHistory WHERE blockheight = ?;"
               [SInt (fromIntegral bCurrent)]
               [RInt]
+        txid <- case r of
+            SInt x -> return x
+            _ -> error "Chainweb.Pact.ChainwebPactDb.handlePossibleRewind.newChildBlock: failed to match SInt"
```
Comment (emilypi) [inline src/Chainweb/Pact/Backend/ChainwebPactDb.hs]: These look like undetected partial patmatches? Yikes 😬

File: cabal.project (hunk 1 above)
Comment (larskuhtz) [reply cabal.project]: 👍 nice

File: src/Chainweb/Pact/Backend/ChainwebPactDb.hs (hunk 2 above)
Comment (larskuhtz) [reply src/Chainweb/Pact/Backend/ChainwebPactDb.hs]: I assume that the result being an `SInt` is a code invariant? This code comes from the db, so there is a serialization step and we'll have to reintroduce the invariant as a type at runtime. So the code is unsafe and we have to do something if our invariant doesn't hold, which would be a bug. I think using `error` is fine in that case.

**Pull request merged**